demo-app/src/main/java/org/openmined/syft/demo/federated/ui/MnistViewModel.kt

```diff
@@ -1,138 +1,89 @@
 package org.openmined.syft.demo.federated.ui
 
-import androidx.lifecycle.MutableLiveData
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.Observer
 import androidx.lifecycle.ViewModel
-import org.openmined.syft.Syft
+import androidx.work.WorkInfo
+import io.reactivex.disposables.CompositeDisposable
+import org.openmined.syft.demo.federated.datasource.LocalMNISTDataDataSource
 import org.openmined.syft.demo.federated.domain.MNISTDataRepository
+import org.openmined.syft.demo.federated.domain.TrainingTask
+import org.openmined.syft.demo.federated.logging.ActivityLogger
+import org.openmined.syft.demo.service.EPOCH
+import org.openmined.syft.demo.service.LOG
+import org.openmined.syft.demo.service.LOSS_LIST
+import org.openmined.syft.demo.service.STATUS
+import org.openmined.syft.demo.service.WorkerRepository
 import org.openmined.syft.domain.SyftConfiguration
-import org.openmined.syft.execution.JobStatusSubscriber
-import org.openmined.syft.execution.Plan
-import org.openmined.syft.execution.SyftJob
-import org.openmined.syft.networking.datamodels.ClientConfig
-import org.openmined.syft.proto.SyftModel
-import java.util.concurrent.ConcurrentHashMap
+import java.util.UUID
 
 private const val TAG = "FederatedCycleViewModel"
 
 @ExperimentalUnsignedTypes
 @ExperimentalStdlibApi
 class MnistViewModel(
-    private val authToken: String,
-    private val configuration: SyftConfiguration,
-    private val mnistDataRepository: MNISTDataRepository
+    private val activity: AppCompatActivity,
```

Comment:
A viewModel should not have a reference to an Android context-based object.

Reply:
If we have to get away with removing activity, then I will also need to shift attachmnistlogger to activity. Essentially there's no need of a viewmodel. Should I shift everything to activity then?

Reply:
Since it's the logger actually behaving as a viewmodel now that needs to be linked to it's lifecycle. May be change logger to be a viewmodel?